### DISCUSSION TITLE
【文章目录】加锁的文章，侧边不显示目录

### DISCUSSION THREAD

**Discussion by PrayerLi** [tangly1024/NotionNext#1974] [Q&A]
**描述bug**
heo主题中，发现加锁的文章点进去之后，不会在侧边显示目录

**复现步骤**
给文章加密码

**期望的正常结果**
枷锁文章也能显示目录

**截图**
未加锁时
![image](https://github.com/tangly1024/NotionNext/assets/87693844/0ed33f63-a90e-45f3-80b0-3abc2b0d9dab)
加锁后
![image](https://github.com/tangly1024/NotionNext/assets/87693844/3a699619-a2c1-4ec3-bc27-6394747ea306)

**环境**

- 操作系统: [windows]
- 浏览器 [edge]
- NotionNext版本 [4.0.12]
- 主题 [例如. heo]

**补充说明**
与问题相关的其它说明

**Comment by duanjr** [comment]
此问题并非heo主题独有的，初步定位问题：
`NotionNext/pages/[prefix]/index.js:56`
```
if (post?.password && post?.password !== '') {
      setLock(true)
    } else {
      setLock(false)
      if (!lock && post?.blockMap?.block) {
        post.content = Object.keys(post.blockMap.block).filter(key => post.blockMap.block[key]?.value?.parent_id === post.id)
        post.toc = getPageTableOfContents(post, post.blockMap)
      }
    }
```
这里在加锁时`post.toc`将为空，导致`SideRight.js`中关于目录的判断有误

**Comment by PrayerLi** [comment]
> 此问题并非heo主题独有的，初步定位问题： `NotionNext/pages/[prefix]/index.js:56`
> 
> ```
> if (post?.password && post?.password !== '') {
>       setLock(true)
>     } else {
>       setLock(false)
>       if (!lock && post?.blockMap?.block) {
>         post.content = Object.keys(post.blockMap.block).filter(key => post.blockMap.block[key]?.value?.parent_id === post.id)
>         post.toc = getPageTableOfContents(post, post.blockMap)
>       }
>     }
> ```
> 
> 这里在加锁时将为空，导致中关于目录的判断有误`post.toc``SideRight.js`

请问大佬有方法修复吗，只略微了解过前端知识，做一些简单的修改还行

**Comment by Serbia580** [comment]
希望大佬修复这个问题，感谢！
